src/nodelets/icp_odometry.cpp

```diff
@@ -307,44 +329,57 @@ class ICPOdometry : public rtabmap_ros::OdometryROS
 		this->processData(data, scanMsg->header.stamp);
 	}
 
-	void callbackCloud(const sensor_msgs::PointCloud2ConstPtr& cloudMsg)
+	void callbackCloud(const sensor_msgs::PointCloud2ConstPtr& pointCloudMsg)
 	{
 		if(this->isPaused())
 		{
 			return;
 		}
+
+		sensor_msgs::PointCloud2 cloudMsg;
+		if (!plugins_.empty())
+		{
+		    cloudMsg = plugins_[0]->filterPointCloud(*pointCloudMsg);
```

Comment:
Add check if the plugin is enabled before calling it. Same for the loop below. 
This could be also included in the loop below with `for(int i=0...`.